python/pyspark/sql/tests.py

```diff
@@ -1937,19 +1937,6 @@ def test_list_functions(self):
             className="org.apache.spark.sql.catalyst.expressions.Add",
             isTemporary=True))
         self.assertEquals(functions, functionsDefault)
-        spark.catalog.registerFunction("temp_func", lambda x: str(x))
-        spark.sql("CREATE FUNCTION func1 AS 'org.apache.spark.data.bricks'")
```

Comment:
Thank you for review, @holdenk .

It's because the current Spark shows the following misleading UX. Users see the registered function by `SHOW FUNCTIONS`, but they also meet `Undefined function` exceptions together. I updated the PR description too.

```
scala> sql("CREATE FUNCTION function_with_invalid_classname AS 'org.invalid'").show
++
||
++
++

scala> sql("show functions like 'function_*'").show(false)
+---------------------------------------+
|function                               |
+---------------------------------------+
|default.function_with_invalid_classname|
+---------------------------------------+

scala> sql("select function_with_invalid_classname()").show
org.apache.spark.sql.AnalysisException: Undefined function: 'function_with_invalid_classname'. This function is neither a registered temporary function nor a permanent function registered in the database 'default'.; line 1 pos 7
```

Also, this is consistent with `CREATE TEMPORARY FUNCTION` behavior, too.